flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/runtime/operators/join/stream/state/JoinRecordStateViews.java

```diff
@@ -79,6 +79,7 @@ static StateTtlConfig createTtlConfig(long retentionTime, boolean stateCleaningE
 				.newBuilder(Time.milliseconds(retentionTime))
 				.setUpdateType(StateTtlConfig.UpdateType.OnCreateAndWrite)
 				.setStateVisibility(StateTtlConfig.StateVisibility.ReturnExpiredIfNotCleanedUp)
+				.cleanupInBackground()
```

Comment:
`cleanupInBackground()` is enabled by default and this method is deprecated. Do we add this in master?
From my understanding, we only need to update 1.9 branch.

Reply:
Yes. I've explained this in the pr's description, and +1 to only fix for 1.9 branch.
Do you mean I need to open the pr against 1.9 branch?

Reply:
Yes. I think we only need a pull request for 1.9 branch.

Reply:
Thanks for your review. I've changed this pr for release-1.9 branch.